docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,621 @@
+> - \>= 1.9: raise this warning to an error
+
+## Standard library
+
+### Warn about potential overload resolution change when `Range`/`Progression` starts implementing `Collection`
+
+> **Issue**: [KT-49276](https://youtrack.jetbrains.com/issue/KT-49276)
+>
+> **Component**: Core language / kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: It is planned to implement `Collection` interface in the standard progressions and concrete ranges
+> inherited from them in Kotlin 1.9. This could make a different overload selected in the overload resolution if there
+> are two overloads of some method, one accepting an element and another accepting a collection.
+> Kotlin will make this situation visible by reporting a warning or an error when such overloaded method is called
+> with a range or progression argument
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning when an overloaded method is called with the standard progression or its range inheritor argument
+>   if implementing `Collection` interface would lead to another overload being selected in this call in future
+> - 1.8.0: raise this warning to an error 
+> - 1.9.0: stop reporting the error, implement `Collection` interface in progressions thus changing
+>   the overload resolution result in the affected cases
+
+### Migrate declarations from `kotlin.dom` and `kotlin.browser` packages to `kotlinx.*`
+
+> **Issue**: [KT-39330](https://youtrack.jetbrains.com/issue/KT-39330)
+>
+> **Component**: kotlin-stdlib (JS)
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: declarations from the `kotlin.dom` and `kotlin.browser` packages are moved to the corresponding `kotlinx.*` packages to prepare for extracting them from stdlib
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: introduce the replacement API in `kotlinx.dom` and `kotlinx.browser` packages
+> - 1.4.0: deprecate the API in `kotlin.dom` and `kotlin.browser` packages and propose the new API above as a replacement
+> - 1.6.0: raise the deprecation level to an error
+> - 1.8.20: remove the deprecated functions from stdlib for JS-IR target
+> - \>= 1.9: move the API in kotlinx.* packages to a separate library
+
+### Deprecate some JS-only API
+
+> **Issue**: [KT-48587](https://youtrack.jetbrains.com/issue/KT-48587)
+>
+> **Component**: kotlin-stdlib (JS)
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: a number of JS-only functions in stdlib are deprecated for removal. They include: `String.concat(String)`, `String.match(regex: String)`, `String.matches(regex: String)`, and the `sort` functions on arrays taking a comparison function, for example, `Array<out T>.sort(comparison: (a: T, b: T) -> Int)`
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: deprecate the affected functions with a warning
+> - 1.9.0: raise the deprecation level to an error
+> - \>=1.10.0: remove the deprecated functions from the public API
+
+## Tools
```

Comment:
Needs Yahor's review from here (I'll use this comment to give it in another place).

Reply:
@Tapchicoma, please, review this file from here.